ethcore/src/trace/db.rs

```diff
@@ -262,6 +262,13 @@ impl<T> TraceDatabase for TraceDB<T> where T: DatabaseExtras {
 	/// Traces of import request's enacted blocks are expected to be already in database
 	/// or to be the currently inserted trace.
 	fn import(&self, batch: &DBTransaction, request: ImportRequest) {
+		// valid (canon):  retracted 0, enacted 1 => false, true,
+		// valid (branch): retracted 0, enacted 0 => false, false,
+		// valid (bbcc):   retracted 1, enacted 1 => true, true,
+		// invalid:	       retracted 1, enacted 0 => true, false,
+		let ret = request.retracted != 0;
+		let ena = !request.enacted.is_empty();
+		assert!((!ret && ena) || (ret == ena));
```

Comment:
Wouldn't `assert!(!(ret && !ena));` work and be more in-line to what is written in comment?
(if I understand correctly there is only one invalid case - `ret == true && ena == false` - so negation will cover all valid cases)

I'm I missing something?